### DISCUSSION TITLE
Tackling package refactorings

### DISCUSSION THREAD

**Discussion by susliko** [scalameta/nvim-metals#291] [Q&A] [answered]
Sometimes one needs to shuffle files or whole directories. While doing this, packages names and references to entities in these packages must be changed

I'm doing this in a cumbersome way:
- move files and directories via [nvim-tree](https://github.com/kyazdani42/nvim-tree.lua) or terminal commands
- live grep lines, which need to be fixed
- add these lines to the quick-fix list and :cdo on them 

Am I missing some handy replacement for drag-n-drop feature of Intellij?

Would be happy to know about alternative workflows tackling such refactorings!

**Comment by ckipp01** [comment, marked as answer]
That's a great questions @susliko. I follow a pretty similar workflow to you where I mostly utilize terminal command to move things around and then refactor afterwards. The killer features that IntelliJ have here is being able to auto-change the package to match your directly structure and also fix imports. In Metals we don't have a great solution for this yet, but it's definitely something we've thought about.

**Reply by susliko** [reply]
Thanks for the reply!
Looking forward to see this feature in Metals!
